HealthVault/Classes/Caching/Private/MHVThingCache.m

```diff
@@ -206,50 +330,49 @@ - (void)prepareCacheForRecords:(MHVRecordCollection *)records
     //Dispatch so app can continue starting while cache is setup
     dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0),^
```

Comment:
Now that these methods are asynchronous, the code within the method implementation should be dispatched on the appropriate queue.